pkg/controller/cluster/controller.go

```diff
@@ -65,18 +66,19 @@ func SetupCluster(mgr ctrl.Manager, l logging.Logger) error {
 		For(&v1alpha1.Cluster{}).
 		Complete(managed.NewReconciler(mgr,
 			resource.ManagedKind(v1alpha1.ClusterGroupVersionKind),
-			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), newArgocdClientFn: cluster.NewClusterServiceClient}),
+			managed.WithExternalConnectDisconnecter(&connectDisconnecter{kube: mgr.GetClient(), newArgocdClientFn: cluster.NewClusterServiceClient}),
 			managed.WithInitializers(managed.NewDefaultProviderConfig(mgr.GetClient())),
 			managed.WithLogger(l.WithValues("controller", name)),
 			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
 }
 
-type connector struct {
```

Comment:
I think we can stick with `connector` as struct name to keep it simple and stick with the naming convention that is used in other providers.

Reply:
Done, I reversed the changes for `connector`